src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -25,14 +25,13 @@ internal static partial class OpenSsl
 
         internal static SafeChannelBindingHandle QueryChannelBinding(SafeSslHandle context, ChannelBindingKind bindingType)
         {
+            Debug.Assert(
+                bindingType != ChannelBindingKind.Endpoint,
+                "Endpoint binding should be handled by EndpointChannelBindingToken");
+
             SafeChannelBindingHandle bindingHandle;
             switch (bindingType)
             {
-                case ChannelBindingKind.Endpoint:
-                    bindingHandle = new SafeChannelBindingHandle(bindingType);
-                    QueryEndPointChannelBinding(context, bindingHandle);
-                    break;
```

Comment:
src/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs 

Endpoint tokens were factored out to share code between macOS and other-Unix.